iwyu_test_util.py

```diff
@@ -503,6 +512,41 @@ def _CheckPrerequisites(cc_file):
     raise unittest.SkipTest(', '.join(failed))
 
 
+def _ParseExpectedFailures(cc_file):
+  """ Parses expected failure conditions out of cc_file. """
+  conditions = []
+  with open(cc_file) as it:
+    for lineno, line in enumerate(it):
+      m = _IWYU_TEST_XFAIL_RE.match(line)
+      if not m:
+        continue
+
+      feature = m.group(1)
+      value = m.group(2)
+
+      if feature is None:
+        # Bare IWYU_XFAIL, condition is always True
+        condition = lambda: True
+        desc = 'always'
```

Comment:
Since we don't have a way to communicate the reason for XFAIL, maybe we should just not produce the desc here? Then you can just return a list of conditions and don't have to ignore the desc when evaluating?

Reply:
Yeah, happy to do that. I wasn't sure if we might want to share the code handling prerequiesites and expected failures so left in the descriptions.

Reply:
Yeah, I think trying to merge them will just make them more complex, but there may be opportunities. Let's remove the desc for now, and get this merged, and we can refactor if we see opportunities later.